modules/mediawiki/manifests/php.pp

```diff
@@ -1,47 +1,185 @@
+    ]
+
+    $core_extensions.each |$extension| {
+        php::extension { $extension:
+            package_name => "php${php_version}-${extension}"
         }
     }
+    # Extensions that are installed with package-name php-$extension and, based
+    # on the php version selected above, will install the proper extension
+    # version based on apt priorities.
+    # php-luasandbox and  php-wikidiff2 are special cases as the package is *not*
+    # compatible with all supported PHP versions.
+    # Technically, it would be needed to inject ensure => latest in the packages,
+    # but we prefer to handle the transitions with other tools than puppet.
+    php::extension { [
+        'apcu',
+        'msgpack',
+        'redis',
+        'luasandbox',
+        'wikidiff2',
+    ]:
+        ensure => present
+    }
+
+    # Extensions that require configuration.
+    php::extension {
+        'xml':
+            package_name => "php${php_version}-xml",
+            priority     => 15;
+        'memcached':
+            priority => 25,
+            config   => {
+                'extension'                   => 'memcached.so',
+                'memcached.serializer'        => 'php',
+                'memcached.store_retry_count' => '0'
+            };
+        'igbinary':
+            config   => {
+                'extension'                => 'igbinary.so',
+                'igbinary.compact_strings' => 'Off',
+            };
+        'mysqli':
+            package_name => "php${php_version}-mysql",
+            config       => {
+                'extension'                 => 'mysqli.so',
+                'mysqli.allow_local_infile' => 'Off',
+            }
+            ;
 
-    $profiling_ensure = $use_tideways ? {
+        'dba':
+            package_name => "php${php_version}-dba",
+    }
+
+    # Additional config files are needed by some extensions, add them
+    # MySQL
+    php::extension {
+        default:
+            package_name => '',;
+        'pdo_mysql':
+            ;
+        'mysqlnd':
+            priority => 10,
+    }
+    # XML
+    php::extension{ [
+        'dom',
+        'simplexml',
+        'xmlreader',
+        'xmlwriter',
+        'xsl',
+    ]:
+        package_name => '',
+    }
+
+    ### FPM configuration
+    # You can check all configuration options at
+    # http://php.net/manual/en/install.fpm.configuration.php
+    if $enable_fpm {
+        class { 'php::fpm':
+            ensure => present,
+            config => {
+                'emergency_restart_interval'  => '60s',
+                'emergency_restart_threshold' => $facts['virtual_processor_count'],
+                'process.priority'            => -19,
+            }
+        }
+
+        # This will add an fpm pool listening on port 8000
+        # We want a minimum of 4 workers
+        $num_workers = max(floor($facts['virtual_processor_count'] * $fpm_workers_multiplier), 4)
+        php::fpm::pool { 'www':
+            port   => 8000,
```

Comment:
Port 8080? We use unix sockets as this would be slower.

Reply:
Fixed. Thanks!